upgrade.go

```diff
@@ -458,8 +458,7 @@ func (b *providerUpgradeBuilder) checkProviderUpgradePreviewOnly(t *testing.T) {
 
 	// Skip if state not yet created
 	if _, err := os.Stat(info.stateFile); os.IsNotExist(err) {
-		t.Skipf("No pre-recorded state found for %s. Run in 'snapshot' mode to capture the current state produced by the current provider version.", b.baselineVersion)
-		return
+		b.providerUpgradeRecordBaselines(t)
```

Comment:
Ah so record right away when baselines are missing? Good idea, it just can take an unexpectedly long time. A log statement is in order. If b.providerUpgradeRecordBaselines(t) logs what it's doing we're good.